embedded-nal-async/src/lib.rs

```diff
@@ -14,4 +14,4 @@ pub use no_std_net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, SocketAddrV4, Socke
 
 pub use dns::Dns;
 pub use embedded_nal::AddrType;
-pub use stack::{TcpClientStack, TcpFullStack, UdpClientStack, UdpFullStack};
+pub use stack::{TcpClient, TcpClientStack, TcpFullStack, UdpClientStack, UdpFullStack};
```

Comment:
If `TcpClient` is successful, I don't think we need the `*Stack` implementations at all?
Further, we'll probably need `UdpClient` trait, as well as `TcpServer` and possibly `UdpServer` traits, whereas the latter two implement socket `bind` + `accept` logic. With multiple connections again, at least for the `accept` part.

Reply:
Yeah, I think they should be kept around until we find an approach that could work for the server side.